### PR TITLE
Shadow light

### PR DESCRIPTION
**Pull request by gazuntype**
allow artist toggle default dynamic lighting on or off
on load, set urp asset to plugin version
Screenshot 2022-08-01 at 18 58 15

before this is merged, new version of app should be available.

### CODE REVIEW

**Review by chornesays — APPROVED** [review]
A few notes. Looks pretty dang good though!

**Comment by chornesays** [comment]
I have one extra thought -- should this be under "Extra settings" as a checkbox instead of its own entry? We have kinda set up the expectation that anything with its own entry is a "simulator only" thing vs. extra settings which is a filter setting.

File: Assets/Shaders/UnlitTextureShadows.shader [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                color *= mainLight.shadowAttenuation;
+                return color;
+            }
+            ENDHLSL
+        }
+
+        // Used for rendering shadowmaps
+        // TODO: there's one issue with adding this UsePass here, it won't make this shader compatible with SRP Batcher
+        // as the ShadowCaster pass from Lit shader is using a different UnityPerMaterial CBUFFER. 
+        // Maybe we should add a DECLARE_PASS macro that allows to user to inform the UnityPerMaterial CBUFFER to use?
```
[the remark this answers is not in the document]
Comment (gazuntype) [reply Assets/Shaders/UnlitTextureShadows.shader:78]: it's from somewhere else. i downloaded the shader. didn't write it myself.

File: Editor/Simulator/SimulatorBase.cs [hunk 2]
```diff
@@ -34,9 +34,15 @@ public enum SimulatorType {
...
     protected Camera mainCamera;
+
+    [SerializeField]
+    protected GameObject defaultLight;
     
     [NonSerialized]
     public bool isPlaying;
+
+    [NonSerialized]
+    public bool dynamicLightOn;
```
[the remark this answers is not in the document]
Comment (gazuntype) [reply Editor/Simulator/SimulatorBase.cs:45]: nah. we default to whatever is already in the user's scene. OnEnable, it checks if the light object is on and defaults to that.

**Comment by gazuntype** [comment]
> I have one extra thought -- should this be under "Extra settings" as a checkbox instead of its own entry? We have kinda set up the expectation that anything with its own entry is a "simulator only" thing vs. extra settings which is a filter setting.

i was thinking of it being in extra settings too. but my problem with it being a checkbox is that it isn't just a setting. it also changes the lighting in the scene. i feel like a checkbox might not really convey that well. dk. it's like the face visualiser button being a checkbox. i'm okay with either sha.

**Comment by chornesays** [comment]
im asking hafid about his expectations. imma guess it should end up as a checkbox

**Comment by chornesays** [comment]
Let's rename it "Environmental Lighting Estimation" and make it a checkbox

I asked him a million questions and that gave the best answers haha

**Pull request merged**